rust/app/src/parser/operations/endorsement.rs

```diff
@@ -160,7 +160,12 @@ impl<'b> EndorsementWithSlot<'b> {
         }
 
         let (rem2, endorsement) = Endorsement::from_bytes(rem)?;
-        let length = (length as usize) - 32 - 1 - (rem.len() - rem2.len());
+        let length = (length as usize).checked_sub(32 + 1);
+        let rem_len = rem.len().checked_sub(rem2.len());
```

Comment:
I think `rem2.len()` will never exceed `rem`, due to `rem2` coming from `rem` or a subslice of it, thus we can avoid this `checked_sub`